gammapy/datasets/map.py

```diff
@@ -337,34 +337,53 @@ def data_shape(self):
         """Shape of the counts or background data (tuple)"""
         return self._geom.data_shape
 
-    def _energy_range(self, mask = None):
-        """Compute the energy range with or without the mask fit."""
-        energy = self._geom.axes["energy"].edges
-        energy_min, energy_max = energy[:-1], energy[1:]
+    def _energy_range(self, mask=None):
+        """Compute the energy range maps with or without the fit mask."""
+        geom = self._geom
+        energy = geom.axes["energy"].edges
+        idx = geom.axes.index_data("energy")
+        geom = geom.drop("energy")
 
         if mask is not None:
-            if mask.data.any():
-                mask = mask.data.any(axis=(1, 2))
+            mask = mask.data
```

Comment:
Is not a bit strange to use the same variable name for 2 different objects?

Reply:
Agreed one could maybe use `_energy_range(mask_map)` or similar...